AppCenter/AppCenter/Internals/Util/MSJwtClaims.m

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSJwtClaims.h"
+#import "MSAppCenterInternal.h"
+#import "MSLogger.h"
+#import <Foundation/Foundation.h>
+
+static NSString *const kMSJwtPartsSeparator = @".";
+static NSString *const kMSSubjectClaim = @"sub";
+static NSString *const kMSExpirationClaim = @"exp";
+
+@implementation MSJwtClaims
+
+- (instancetype)initWithSubject:(NSString *)subject expirationDate:(NSDate *)expirationDate {
+  self = [super init];
+  if (self) {
+    _subject = subject;
+    _expirationDate = expirationDate;
+  }
+  return self;
+}
+
++ (MSJwtClaims *)parse:(NSString *)jwt {
+  NSArray *parts = [jwt componentsSeparatedByString:kMSJwtPartsSeparator];
+  if ([parts count] < 2) {
+    MSLogError([MSAppCenter logTag], @"Failed to parse JWT, not enough parts.");
+    return nil;
+  }
+  NSString *base64ClaimsPart = parts[1];
+  @try {
+    NSError *error;
+    NSData *claimsPartData = [[NSData alloc] initWithBase64EncodedString:base64ClaimsPart options:0];
+    NSDictionary *claims = [NSJSONSerialization JSONObjectWithData:claimsPartData options:0 error:&error];
+    if ([claims objectForKey:kMSExpirationClaim] == nil || [claims objectForKey:kMSSubjectClaim] == nil) {
+      MSLogError([MSAppCenter logTag], @"Deserialized claims is missing `sub` or `exp`");
+      return nil;
+    }
+
+    // We will treat invalid expiration times as an interval of 0 because there is not an easy way to determine the
```

Comment:
Just FYI, this is a behavior difference from Android.  Does anyone have feedback on why this could be problematic?  Or is it OK?

Reply:
@achocron Android is able to actually throw in the try/catch 
https://github.com/microsoft/appcenter-sdk-android/pull/1240/files#diff-c86e6f6d9a168af1ce6ac45f63d50114R57. You can see in our test [here](https://github.com/microsoft/appcenter-sdk-apple/pull/1753/files#diff-23f775a2e0f4731debe8611846ecb5bfR84) that we return a claim with an old interval, android is able to return empty claims [here](https://github.com/microsoft/appcenter-sdk-android/pull/1240/files#diff-43f6e0c4ac3bab52fe59f69bcf67988bR67)